Database setup

jpwalbran opened the pull request:
Merging the database configuration into main. (This includes the database configuration itself, and all of the server side functionality to communicate with the database)

jpwalbran committed:
Seed database for the product/pantry schema, and started server implementation

Implemented the database schema as discussed as a group, found in the #project-discussion slack channel. Additionally worked on server infrastructure to work with the new collections.

jpwalbran committed:
Started work on ProductController.java

Finished working on the key properties for the Pantry interface, as well as fixed the regex pattern for company ignorecase in users. Primarily started working on ProductController.java, committing changes before merging from main after PR.

jpwalbran committed:
Merge branch 'main' into database-setup

jpwalbran committed:
Fixed VSCode errors

Finished preliminary cleanup after merging main into the branch, pushing to see if it resolves github check wierdness, and also just pushing results at the end of a work session.

jpwalbran committed:
Add tests for ProductController.spec.java

Started to add tests to the Product Controller on the server side. The coverage is currently under the required threshold, and two of the three tests are failing. (canGetAllProducts, and responds correctlytoError) Committing now for end of lab.

OllieWillette667 reviewed (commented):
I am confused, I already did this part

jpwalbran committed:
Changed productName field to adhere with linting

Changed naming to be consistent with checkStyle, and commiting to change machines.

Shua59 committed:
Finish tests on ProductController.java

Finish implementing tests on the ProductController.spec.java, and ran against checkStyle. 
Co-Authored-By: jpwalbran <walbr042@morris.umn.edu>

Shua59 committed:
Add tests to ProductControllerSpec to increase coverage

Shua59 committed:
Merge branch 'main' into database-setup
